eslint/babel-eslint-parser/src/configuration.js

```diff
@@ -9,7 +9,19 @@ export function normalizeESLintConfig(options) {
     requireConfigFile: true,
   };
 
-  return Object.assign(defaultOptions, options);
+  // ESLint sets ecmaVersion: undefined when ecmaVersion is not set in the config.
+  // Prune to ensure that defaults are respected.
+  const prunedOptions = Object.entries(options).reduce(
+    (options, [key, value]) => {
+      if (value !== undefined) {
+        options[key] = value;
+      }
+      return options;
+    },
+    {},
+  );
+
+  return Object.assign(defaultOptions, prunedOptions);
```

Comment:
What do you think avoiding `prunedOptions` and `Object.assign`?

```js
  const {
    babelOptions = {},
    ecmaVersion = 2020,
    sourceType = "module",
    allowImportExportEverywhere = false,
    requireConfigFile = true,
    ...otherOptions
  } = options;

  return {
    babelOptions,
    ecmaVersion,
    sourceType,
    allowImportExportEverywhere,
    requireConfigFile,
    ...otherOptions
  };
```

Reply:
Sure!